utilities/merge_operators/string_append/stringappend2.cc

```diff
@@ -51,14 +53,14 @@ bool StringAppendTESTOperator::FullMergeV2(
     printDelim = true;
   } else if (numBytes) {
     merge_out->new_value.reserve(
-        numBytes - 1);  // Minus 1 since we have one less delimiter
+        numBytes - delim_.size());  // Adjust for one less delimiter
```

Comment:
Need a simple explanation for why we minus delim_.size()

Reply:
We want to reserve the space for the original value and N following operands with preceding delimiters. The size of "N following operands with preceding delimiters" is stored as `numBytes`. However, if there is no original value, the very first delimiter becomes redundant, therefore we reduce the amount of space needed by the size of one delimiter. 

For instance, if the original value was 'foo' and operands 'bar', 'baz' with delimiter ',', we would need `3 + numBytes = 3 + 2 * (3 + 1) = 11 bytes`, however if there is no original value, we need only `0 + numBytes - 1 = 7 bytes`.
This adjustment just takes into account that a delimiter can now be of an arbitrary length, not only one byte.

Reply:
Thanks! Can you add these as the comment to the code?